File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/rules/RecordNeedsManualTriage.java
package com.freight.rules;

/** Triage predicate: record needs manual triage. */
public final class RecordNeedsManualTriage implements TriageRule {

    @Override
    public String name() {
        return "record_needs_manual_triage";
    }

    @Override
    public boolean apply(ProbeRecord record) {
        return new HazmatRequiresEscort().apply(record) || new MassExceedsSoftCap().apply(record) || !new LaneIndexInRange().apply(record);
    }
}

File: freight-triage-polyglot-ledger-recovery/environment/reconcile/internal/rulex/record_needs_manual_triage.go
package rulex

// RecordNeedsManualTriage is the triage predicate for record needs manual triage.
func RecordNeedsManualTriage(record Record) bool {
	return HazmatRequiresEscort(record) || MassExceedsSoftCap(record) || !LaneIndexInRange(record)
}
